pkg/kube/wait.go

```diff
@@ -221,6 +231,14 @@ func (w *waiter) volumeReady(v *corev1.PersistentVolumeClaim) bool {
 	return true
 }
 
+func (w *waiter) storageClassReady(sc *storagev1beta1.StorageClass) bool {
+	if len(sc.Provisioner) <= 0 {
```

Comment:
I was looking at the reference docs and documentation for storage classes and I am not sure this is the best field to be keying off of. It looks like `provisioner` is a required field anyway and so this is more of a validation issue. However, I can't seem to figure out what field would need to be used. Is there something better we can key off of to solve the problem as described in #5958?

Reply:
In the given issue the problem is taking place with CRDs not being instantiated properly. And is the issue with the APIs which the storageos operator is trying to expose. The storageclasses defined over there are provided by the storageos apis and not something supported here. Here we are mainly trying to listen on StorageClasses defined in Kubernetes.

Reply:
I have discussed with a few folks working in storage, more specifically Ceph. It seems that the check the health of the storage class is based on if the provisioner is up. In #5958 @staticdev is trying to install an operator which upon installing exposes the storageos.io/v1

Reply:
If I am understanding it correctly, this is trying to watch for storage classes that are created by the operator, which are completely out of scope from Helm itself. This feels like something that should be done in a `post-install` hook rather than within the wait logic itself

Reply:
This PR watches StorageClasses defined via Kubernetes by default. 

The listening on CustomResources such as StorageOS StorageClasses is something that would be part of a larger CRD integration Story as discussed here https://github.com/helm/helm/issues/5871 is what I feel. This PR would not be catering to that, but the Users should be able to listen on successful creation of StorageClasses nevertheless.

Reply:
So I don't think it will work as defined in this PR though. The wait logic only waits for resources that were defined as part of the release and not the default resources that are already there. I am also still not sure what constitutes a storage class being ready based on all of my past experience and current reading of the docs. As I stated before, according to the spec, `provisioner` is a required field, so basically all we are doing is retrieving the object from the k8s API to see if we can do it. Is there anything that Helm itself can "wait" on that isn't a specific provider of a storage class?

It may be valuable for us to sync up over a call so I make sure I am not missing anything